cmake/CodeTidy.cmake

```diff
@@ -0,0 +1,42 @@
+IF (NOT CLANG_TIDY_EXECUTABLE)
+  FIND_PROGRAM(CLANG_TIDY_EXECUTABLE
+    NAMES
+    clang-tidy-5.0
+    clang-tidy-mp-5.0
+    clang-tidy
+  )
+  IF (CLANG_TIDY_EXECUTABLE)
+    MESSAGE("-- Found clang-tidy " ${CLANG_TIDY_EXECUTABLE})
+  ELSE()
+    MESSAGE(FATAL_ERROR "-- clang-tidy not found")
```

Comment:
We use `SEND_ERROR` instead of `FATAL_ERROR`

Reply:
Sure.  Want me two change all of these occurrences?
```bash
$ git grep FATAL_ERROR ..
../cmake/CodeCoverage.cmake:  MESSAGE(FATAL_ERROR "-- lcov not found")
../cmake/CodeCoverage.cmake:  MESSAGE(FATAL_ERROR "-- genhtml not found")
../cmake/CodeFormat.cmake:    MESSAGE(FATAL_ERROR "-- clang-format not found")
../cmake/CodeFormat.cmake:    MESSAGE(FATAL_ERROR "-- clang-format path is invalid")
../cmake/CodeFormat.cmake:    MESSAGE(FATAL_ERROR "You must use clang-format version 5.0")
../cmake/CodeTidy.cmake:    MESSAGE(FATAL_ERROR "-- clang-tidy not found")
../cmake/CodeTidy.cmake:    MESSAGE(FATAL_ERROR "-- clang-tidy path is invalid")
../cmake/CodeTidy.cmake:    MESSAGE(FATAL_ERROR "You must use clang-tidy version 5.0")
```

Reply:
Yes, thanks